libs/storage/src/ipfs/ipfs.service.ts

```diff
@@ -70,6 +73,17 @@ export class IpfsService {
     return response;
   }
 
+  public async checkExists(cid: string): Promise<boolean> {
```

Comment:
is this safe to use from durability perspective since we might have files that do exist but are not pinned!?

Reply:
Hmmm... for the `content-publishing` use case, at least, we expect that any file we're querying was recently uploaded by the same content-publishing instance to the same IPFS node, so if the local node doesn't have it, it probably doesn't exist.

However, you bring up a good point for naming, at least--perhaps:
```suggestion
  public async checkExistsInLocalGateway(cid: string): Promise<boolean> {
```

Reply:
I might be getting this completely wrong but my understanding is that there exists a state that you uploaded a file but that upload would be only temporary unless you pin it. So if this assumption is correct then is there a possibility that you would find that file locally but it's not pinned in reality and it can get removed by node at any minute.

Reply:
* For content-publishing, we don't just "upload", we pin. Though, we don't "care" if it's pinned (though it makes sense for it to be so; we only need to verify that it was uploaded & available currently). Whether any content remains pinned after that is not guaranteed; ie, as a provider I could remove pins whenever I want to manage my IPFS resources.
* For content-watcher, we don't care if it's pinned at all, just that it's available at the time we discover it.

So, for content-publishing purposes, we upload & pin (because we want the content to remain available), but from a workflow perspective we are mainly interested that the upload was successful & the resource is currently available. Because even if it's pinned right now, that pin could be removed at any point that the owner of the node so chooses.